Indicators/Correlation.cs

```diff
@@ -0,0 +1,194 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.Indicators
+{
+    /// <summary>
+    /// The Correlation Indicator is a valuable tool in technical analysis, designed to quantify the degree of 
+    /// relationship between the price movements of a target security (e.g., a stock or ETF) and a reference 
+    /// market index. It measures how closely the target’s price changes are aligned with the fluctuations of 
+    /// the index over a specific period of time, providing insights into the target’s susceptibility to market 
+    /// movements.
+    /// A positive correlation indicates that the target tends to move in the same direction as the market index, 
+    /// while a negative correlation suggests an inverse relationship. A correlation close to 0 implies a weak or 
+    /// no linear relationship.
+    /// Commonly, the SPX index is employed as the benchmark for the overall market when calculating correlation, 
+    /// ensuring a consistent and reliable reference point. This helps traders and investors make informed decisions 
+    /// regarding the risk and behavior of the target security in relation to market trends.
+    /// </summary>
+
+    public class Correlation : BarIndicator, IIndicatorWarmUpPeriodProvider
```

Comment:
Thank you for upgrading from `TradeBarIndicator` to `BarIndicator`
On this PR, we added a test for accepting `QuoteBar` which will assert this change:
https://github.com/QuantConnect/Lean/pull/7540/files#diff-0ad130eeee282f4dfc889caec0e1f31988ca69550f89eed6fb86e15ec5517995R147

Reply:
just pushed changed with AcceptsQuoteBarsAsInput test